api/src/test/java/org/openmrs/validator/ObsValidatorTest.java

```diff
@@ -511,4 +524,84 @@ public void validate_shouldPassValidationIfValueTextIsLessThanTheMaximumLength()
 		assertFalse(errors.hasFieldErrors("obsDatetime"));
 		assertFalse(errors.hasFieldErrors("valueText"));
 	}
+
+	/**
+	 * @see ObsValidator#validate(java.lang.Object, org.springframework.validation.Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfObsValueExceedsHiAbsolute() {
+		Obs obs = new Obs();
+		obs.setId(1);
+		obs.setPerson(new Person(10));
+		obs.setConcept(Context.getConceptService().getConcept(4090));
+		obs.setValueNumeric(200.0);
+		obs.setObsDatetime(new Date());
+		
+		Errors errors = new BindException(obs, "obs");
+		obsValidator.validate(obs, errors);
+
+		assertTrue(errors.hasFieldErrors("valueNumeric"));
+	}
+
+	/**
+	 * @see ObsValidator#validate(java.lang.Object, org.springframework.validation.Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfObsValueBelowLowAbsolute() {
+		Obs obs = new Obs();
+		obs.setPerson(new Person(10));
+		obs.setConcept(Context.getConceptService().getConcept(4089));
+		obs.setValueNumeric(50.0);
+		obs.setObsDatetime(new Date());
+		
+		Errors errors = new BindException(obs, "obs");
+		obsValidator.validate(obs, errors);
+
+		assertTrue(errors.hasFieldErrors("valueNumeric"));
+	}
+
+	/**
+	 * @see ObsValidator#validate(java.lang.Object, org.springframework.validation.Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfObsAgeNotInRange() {
+		Person person = new Person();
+		calendar.set(1900, Calendar.JANUARY, 1);
+		person.setBirthdate(calendar.getTime());
+		
+		Obs obs = new Obs();
+		obs.setId(1);
+		obs.setPerson(person);
+		obs.setConcept(Context.getConceptService().getConcept(4090));
+		obs.setValueNumeric(140.0);
+		obs.setObsDatetime(new Date());
+
+		Errors errors = new BindException(obs, "obs");
+		obsValidator.validate(obs, errors);
+
+		assertTrue(errors.hasFieldErrors("valueNumeric"));
+		assertNotNull(errors.getFieldError("valueNumeric"));
+		assertEquals("error.outOfRange.criteria.not.match", errors.getFieldError("valueNumeric").getCode());
+	}
+	
+	/**
+	 * @see ObsValidator#validate(java.lang.Object, org.springframework.validation.Errors)
+	 */
+	@Test
+	public void validate_shouldPassValidationIfObsWithinRange() {
+		Obs obs = new Obs();
+		Person person = new Person(1);
```

Comment:
Can we get rid of the variables that we use just once?

Reply:
I am setting birthdate below